std/net/curl.d

```diff
@@ -627,6 +627,60 @@ unittest
     assert(res == cast(ubyte[])[17, 27, 35, 41]);
 }
 
+/* Encode associative array using www-form-urlencoding
+ * perhaps this should live in std.uri
+ */
+private string urlEncode()(in string[string] values)
+{
+    bool first = true;
+    Appender!string enc;
+    enc.reserve(values.length * 512);
+    foreach (k, v; values)
+    {
+        if (!first)
+            enc ~= '&';
+        import std.uri : encodeComponent;
+
+        enc ~= encodeComponent(k);
+        enc ~= "=";
+        enc ~= encodeComponent(v);
+        first = false;
+    }
+    return enc.data;
+}
+
+/** HTTP post content.
+ *
+ * Params:
+ * url = resource to post to
+ * postData = data to send as the body of the request. An associative array of $(D string) is accepted and will be
+ *            encoded using www-form-urlencoding
+ * conn = HTTP connection to use
+ *
+ * Returns:
+ * A $(D char[]) containing the content of the resource pointed to by the URL.
+ *
+ * See_Also: $(LREF HTTP.Method)
+ */
+char[] post()(in string url, in string[string] postData, HTTP conn = HTTP())
+{
+    return post(url, urlEncode(postData), conn);
+}
+
+///
+unittest
+{
+    import std.net.curl;
+
+    auto content = post("d-lang.appspot.com/testUrl2", ["name1" : "value1", "name2" : "value2"]);
```

Comment:
Please inline this example like as in the other `post` test.
Unittest are sometimes run on _isolated_ machines without internet connection. It would break then :/